### DISCUSSION TITLE
How to avoid fetching files located in subdirectories ? (option `deep: false` in v1)

### DISCUSSION THREAD

**Discussion by benoitdemaegdt** [nuxt/content#1777] [Q&A] [answered]
Hey there 👋 

Does anyone know if there is a way to avoid fetching files located in subdirectories when querying with nuxt/content ?

Let's say I have the following structure in the `content` folder
```
├── blog
│   ├── category-1
│   │   ├── article-1-in-category-1.json
│   ├── article-1.md
│   ├── article-2.md
│   ├── article-3.md
```

If my query in `pages/blog/index.vue` looks like this 
```
const { data: articles } = await useAsyncData(() => {
  return queryContent('blog').find()
})
```
I'll get `article-1`, `article-2`, `article-3` **and** `article-1-in-category-1`.

However this `article-1-in-category-1` is in a subdirectory because the content is much different (json vs md in my case).

Is there a way not to fetch files in subdirectories ?
Nuxt/content V1 provided a [very handy deep option](https://content.nuxtjs.org/v1/getting-started/fetching/#contentpath-options) but I cannot find it in Nuxt V2 doc.

<img width="345" alt="image" src="https://user-images.githubusercontent.com/25749578/209638159-54ce3e94-be84-4201-92cc-6cd7f99d4cb5.png">

**Comment by farnabaz** [comment, marked as answer]
You can achieve your goal using RegExp matching.
```ts
const { data: articles } = await useAsyncData(() => {
  return queryContent().where({ _path: /^\/blog\/[^/]*$/ }).find()
})
```

**Reply by benoitdemaegdt** [reply]
I had to ask ChatGPT for understanding this regex, but it works fine 👌 

Thank you @farnabaz  😃 

<img width="829" alt="Capture d’écran 2022-12-27 à 17 48 40" src="https://user-images.githubusercontent.com/25749578/209697129-8038c6ef-e694-4078-a2a7-cc5f701b198b.png">
